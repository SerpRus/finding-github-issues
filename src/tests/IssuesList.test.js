import { mount } from '@vue/test-utils';
import issuesList from '../components/issuesList.vue';

describe('issuesList tests', () => {
  const data = [{
    html_url: 'https://github.com/vuejs/vue/pull/12737',
    number: 12737,
    state: 'open',
    created_at: '2022-08-11T09:27:06Z',
    title: 'feat(types): new Vue() improvements (#12730)',
    user: {
      login: 'gulewei',
      html_url: 'https://github.com/gulewei',
    },
  },
  {
    html_url: 'https://github.com/vuejs/vue/issues/12708',
    number: 12708,
    state: 'open',
    created_at: '2022-07-27T20:41:30Z',
    title: '::v-deep is not working',
    user: {
      login: 'handhikadj',
      html_url: 'https://github.com/handhikadj',
    },
  }];

  it('correct issuesData', () => {
    const wrapper = mount(issuesList, {
      props: {
        issuesData: data,
      },
    });

    expect(wrapper.findAll('.issue-item').length).toBe(data.length);
  });

  it('No issuesData', () => {
    const wrapper = mount(issuesList, {
      props: {
        issuesData: [],
      },
    });

    expect(wrapper.findAll('.issue-item').length).toBe(0);
  });

  it('issuesList current data snapshot', () => {
    const wrapper = mount(issuesList, {
      props: {
        issuesData: data,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('issuesList no data snapshot', () => {
    const wrapper = mount(issuesList, {
      props: {
        issuesData: [],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
