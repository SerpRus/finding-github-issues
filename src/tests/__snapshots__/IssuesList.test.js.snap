// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issuesList tests issuesList current data snapshot 1`] = `
"<ul id=\\"issues-list\\" class=\\"issues-list\\">
  <li class=\\"issue-item\\">
    <div class=\\"issue-item__state\\"> State: <span class=\\"issue-item__state-value\\">open</span></div><a class=\\"issue-item__link\\" href=\\"https://github.com/vuejs/vue/pull/12737\\">feat(types): new Vue() improvements (#12730)</a>
    <div class=\\"issue-item__info\\">
      <div class=\\"issue-item__number-with-date\\">
        <div class=\\"issue-item__number\\"> #12737</div>
        <div class=\\"issue-item__date\\"> opened on 11 Aug</div>
      </div>
      <div class=\\"issue-item__author\\"> by <a class=\\"issue-item__login\\" href=\\"https://github.com/gulewei\\">gulewei</a></div>
    </div>
  </li>
  <li class=\\"issue-item\\">
    <div class=\\"issue-item__state\\"> State: <span class=\\"issue-item__state-value\\">open</span></div><a class=\\"issue-item__link\\" href=\\"https://github.com/vuejs/vue/issues/12708\\">::v-deep is not working</a>
    <div class=\\"issue-item__info\\">
      <div class=\\"issue-item__number-with-date\\">
        <div class=\\"issue-item__number\\"> #12708</div>
        <div class=\\"issue-item__date\\"> opened on 27 Jul</div>
      </div>
      <div class=\\"issue-item__author\\"> by <a class=\\"issue-item__login\\" href=\\"https://github.com/handhikadj\\">handhikadj</a></div>
    </div>
  </li>
</ul>"
`;

exports[`issuesList tests issuesList no data snapshot 1`] = `"<!--v-if-->"`;
